### DISCUSSION TITLE
How to include columns associated with Virtual classes in audit entry.

### DISCUSSION THREAD

**Discussion by supr-pr** [thepirat000/Audit.NET#661] [Q&A] [answered]
[AuditInclude]
public class ProgramCategory : EntityBase
{
[AuditIgnore]
public Guid Id { get; set; }
public string Name { get; set; }
[AuditIgnore]
public string CreatedBy { get; set; }
[AuditIgnore]
public string UpdatedBy { get; set; }
public bool IsDeleted { get; set; }
}
 
[AuditInclude]
public class Program : EntityBase, IHaveEncryptedProperties
{
 
public Program() { }
 
[AuditIgnore]
public long Id { get; private set; }
[AuditIgnore]
[Encrypt]
public string Name { get; protected set; }
public string NameDecrypted { get; private set; }
. . .
public virtual ProgramCategory ProgramCategory { get; set; }
}
 
I am trying to audit the Program Entity
I need the value of the key for ProgramCategory in the audit whenever i change its values.
 
In DB, it corresponds to the column [ProgramCategory_Id]
 
SELECT TOP (1000) [Id]
,[Name]
,[NameDecrypted]
. . .
,[ProgramCategory_Id]
FROM [cwm].[dbo].[Programs]
 
But i cannnot seem to get it in the audit records.
 
 
{
"schema": "dbo",
"table": "Programs",
"primaryKey": {
"id": "10022"
},
"action": "Update",
"changes": [
{
"columnName": "NameDecrypted",
"originalValue": "asdasdad",
"newValue": "Let Us Always Change This"
}
],
"columnValues": {
"nameDecrypted": "Let Us Always Change This",
"deleted": false,
"modifiedUtc": "2024-03-25T06:51:28.6647371Z",
"createdUtc": "2023-08-24T09:49:34.563",
. . .
},
"valid": true,
"validationResults": []
}

**Comment by thepirat000** [comment, marked as answer]
Please include the libraries/versions you're referencing, the net framework version you're targeting and your configuration code for audit.net.

Do you have a mapping to the ProgramCategory_Id foreign key mapped to your program entity?, If not, can you try adding the property?

https://stackoverflow.com/questions/60663539/ef-core-should-i-add-foreign-key-id-in-class

```
public class Program : EntityBase, IHaveEncryptedProperties
{
    // ...
    public int ProgramCategoryId { get; set; }
}
```

**Reply by supr-pr** [reply]
Oh! Yes, did just that. Apologies, forgot to mention here.
Thanks!
